app/src/main/kotlin/com/wire/android/ui/home/conversations/model/MessageTypes.kt

```diff
@@ -131,10 +135,9 @@ fun MessageButtonsContent(
         for (index in buttonList.indices) {
             val button = buttonList[index]
             MessageButtonItem(
-                messageId = messageId,
-                pendingButtonId = pendingButton,
+                pendingButtonId = viewModel.pendingButtonId,
```

Comment:
any reason why you decided not pass here the boolean instead of the id and do the check inside MessageButtonItem ?

Reply:
and the composable will look like this

```kotlin
@Composable
fun MessageButtonItem(
    button: MessageButton,
    isPending: Boolean,
    isAnyPending: Boolean,
    doButtonAction: ((buttonId: String) -> Unit)
) {
.
.
.
.

    val onCLick = remember(button.isSelected) {
        doButtonAction.let {
            if (!button.isSelected) {
                { doButtonAction(button.id) }
            } else {
                { }
            }
        }
    }


val state = if (button.isSelected) WireButtonState.Selected
    else if (isAnyPending) WireButtonState.Disabled
    else WireButtonState.Default
    WireSecondaryButton(
        loading = isPending,
        text = button.text,
        onClick = onCLick,
        state = state
    )

}
```

and i am not sure if this improve readablity

Reply:
sorry but I am getting a bit confused :S I still think the the child composable should not have this logic, I think it would be nicer if it was independent of anything related to id, because in the end, you make it coupled to something that pass that id, and makes it less reusable, but you are right having isAnyPending is also not ideal solution. 

I think a middle ground would be to create a pass the state object to the MessageButtonItem and determine from the parent what state it needs to be composed with. That way we still have less coupled Composable but still are able to pass the correct state from within the parent. 

Anyways this is nothing blocking, and I do want to block you with the PR, I am going to approve it for now, but it would be nice if you atleast try the suyggestion and see it makes sense for you well done 🎖️